### PR TITLE
Added signal triggered before fork

### PR DESCRIPTION
**Pull request by jaroslawporada**
*Note*: Before submitting this pull request, please review our [contributing
guidelines](https://docs.celeryq.dev/en/main/contributing.html).

## Description

Created signal emitted just before creating new child process.
I need to clean some classes which use grpc channels and ssl connections.

In our case GCP secretmanager using grpc channel cause many "Timed out waiting for UP message" due to stuck on command:
Zrzut ekranu z 2023-03-23 17-07-15

### CODE REVIEW

File: docs/userguide/signals.rst [hunk 1]
```diff
@@ -543,6 +543,13 @@ Provides arguments:
 
 Dispatched before the worker is started.
 
+.. signal:: worker_before_create_process
+
+``worker_before_create_process``
+~~~~~~~~~~~~~~~
```
Comment (auvipy) [inline docs/userguide/signals.rst:549]:
```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: celery/signals.py [hunk 2]
```diff
@@ -105,6 +106,7 @@
 # - Worker
 import_modules = Signal(name='import_modules')
 worker_init = Signal(name='worker_init')
+worker_before_create_process = Signal(name="worker_before_create_process")
```
Comment (auvipy) [inline celery/signals.py:109]: would like to have some unit/integration tests to see its working well and won't create any regression

**Review by auvipy — CHANGES_REQUESTED** [review]

File: celery/signals.py (hunk 2 above)
Comment (Nusnus) [reply celery/signals.py:109]: +1

**Comment by jaroslawporada** [comment]
Hi,
I made changes you requested. I kindly ask for another code review.
BR,
Jarek

**Review by auvipy — APPROVED** [review]

**Pull request merged**